src/modules/site-v2/base/views/strains.py

```diff
@@ -143,59 +146,144 @@ def strains_catalog():
     flash(Markup("Strain mapping sets 9 and 10 will not be available until later this year."), category="primary")
     title = "Strain Catalog"
     warning = request.args.get('warning')
-    disable_parent_breadcrumb = True
     strain_listing = get_strains()
     strain_sets = get_strain_sets()
+    form = OrderForm()
     return render_template('strain/catalog.html', **locals())
 
 #
 # Strain Ordering Pages
 #
 
-@strains_bp.route('/checkout', methods=['GET', 'POST'])
+@strains_bp.route('/checkout', methods=['POST'])
 @jwt_required(optional=True)
-def order_page():
-  """ This view handles the order page """
+def order_page_post():
   form = OrderForm()
   user = get_current_user()
-  if user and hasattr(user, 'email') and not form.email.data:
-    form.email.data = user.email
+  cartID = request.cookies.get('cartID')
+  if user:
+    users_cart = Cart.lookup_by_user(user['email'])
+  elif cartID:
+    users_cart = Cart(cartID)
+  else:
+    users_cart = Cart(**{'items': []})
+
+  if 'order_form' in request.form:
+    if form.validate_on_submit():
+      """ submitting the order """
+      cartItems = users_cart['items']
+      if form.shipping_service.data == 'Flat Rate Shipping':
+        cartItems.append({'name': 'Flat Rate Shipping'})
+      for item in cartItems:
+        item_price = users_cart.get_price(item)
+        item['price'] = item_price
+      totalPrice = sum(item['price'] for item in cartItems)
+      
+      # When the user confirms their order it is processed below.
+      order_obj = {'total': totalPrice,
+                    'date': datetime.now(timezone.utc).date().isoformat(),
+                    'is_donation': False}
+      order_obj.update(form.data)
+      order_obj['phone'] = order_obj['phone'].strip("+")
+      order_obj['items'] = '\n'.join(sorted([f"{item['name']}:{item['price']}" for item in cartItems]))
+      order_obj['invoice_hash'] = str(uuid.uuid4()).split("-")[0]
+      order_obj["order_confirmation_link"] = url_for('request_strains.order_confirmation', invoice_hash=order_obj['invoice_hash'], _external=True)
+      send_email({"from": "no-reply@elegansvariation.org",
+                  "to": [order_obj["email"]],
+                  "cc": config.get("CC_EMAILS"),
+                  "subject": "CeNDR Order #" + str(order_obj["invoice_hash"]),
+                  "text": ORDER_SUBMISSION_EMAIL_TEMPLATE.format(**order_obj)})
+
+      # Save to google sheet
+      add_to_order_ws(order_obj)
+
+      # Mark the cart is deleted
+      users_cart.delete_cart()

```

Comment:
@orzut  -- since we are doing a 'soft-delete', we might want to rename this method to #soft_delete(). Also, since the method applies to `Cart`, we can remove the suffix `_cart` from the method name.